src/build_context.rs

```diff
@@ -180,8 +180,10 @@ fn extract_used_refs(schema: &PyAny, refs: &mut AHashSet<String>) -> PyResult<()
         if is_definition_ref(dict)? {
             refs.insert(dict.get_as_req(intern!(schema.py(), "schema_ref"))?);
         } else {
-            for (_, value) in dict.iter() {
-                extract_used_refs(value, refs)?;
+            for (key, value) in dict.iter() {
+                if !key.eq(intern!(schema.py(), "metadata"))? {
```

Comment:
I noticed that right now, it seems a bit dangerous that it could look into the 'metadata' key while checking if something is a defintion ref, and while looking for refs used.

Happy to pull this into a separate PR if you'd prefer. Also can try to add a test of this if appropriate.

Reply:
To be honest I'm not really sure what's supposed to be going on in this file...

Reply:
I'll put this in a separate PR.